handlers_test.go

```diff
@@ -464,19 +465,24 @@ func TestGZIP(t *testing.T) {
 	srv := testServer()
 	defer srv.Close()
 
-	resp, err := http.Get(srv.URL + "/gzip")
+	client := new(http.Client)
+	req, err := http.NewRequest("GET", srv.URL + "/gzip", nil)
+	require.Nil(t, err)
+
+	req.Header.Add("Accept-Encoding", "gzip")
+	resp, err := client.Do(req)
 	require.Nil(t, err)
 	defer resp.Body.Close()
 
-	// net/http.Client removes "Content-Encoding: gzip"
-	// and adds "Content-Type: application/x-gzip" in the
-	// absence of "Content-Type" header.
-	require.EqualValues(t, "application/x-gzip", resp.Header.Get("Content-Type"))
+	require.EqualValues(t, "gzip", resp.Header.Get("Content-Encoding"))
+	require.EqualValues(t, "application/json", resp.Header.Get("Content-Type"))
+	zr, err := gzip.NewReader(resp.Body)
```

Comment:
I noticed that request body stays encoded if response header had Accept-Encoding. So I decoded response body additionally.

When I send request with http.Get(), it removes Content-Encoding header. After the change I made in handler, Content-Type is "application/json". I think it's meaningless if I only check "Content-Type: application/json" in gzip test. 

So I decided to add Accept-Encoding to request header and decode additionally.